analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -92,7 +93,8 @@ private static ProgramState LearnBranchingRelationalNumberConstraint(ProgramStat
         }
 
         ProgramState LearnBranching(ISymbol leftSymbol, BinaryOperatorKind kind, NumberConstraint rightConstraint) =>
-            RelationalNumberConstraint(kind, rightConstraint) is { } leftConstraint
+            !(falseBranch && leftSymbol.GetSymbolType().IsNullableValueType())  // Don't learn opposite for "nullable != 0", because it could also be <null>.
```

Comment:
The comment needs to be corrected. We are looking at relational operators rather than equality.